packages/web/stores/account.ts

```diff
@@ -0,0 +1,58 @@
+import { assets, chain } from "@chain-registry/osmosis";
+import { Logger, WalletManager } from "@cosmos-kit/core";
+import { wallets as keplrWallets } from "@cosmos-kit/keplr";
+import { wallets as leapWallets } from "@cosmos-kit/leap";
+import { action, makeObservable, observable } from "mobx";
+
+const chains = ["osmosis"] as const;
+const logger = new Logger(console, "WARN");
+
+export class AccountStore {
+  @observable
+  private _refreshRequests = 0;
+
+  private _walletManager: WalletManager = new WalletManager(
+    [chain],
+    [assets],
+    [...keplrWallets, ...leapWallets],
+    logger,
+    "icns"
+  );
+
+  constructor() {
+    this.walletManager.walletRepos.forEach((repo) => {
+      repo.wallets.forEach((wallet) => {
+        wallet.setActions({
+          state: () => this.refresh(),
+        });
+      });
+    });
+
+    makeObservable(this);
+
+    this.walletManager.onMounted();
+  }
+
+  @action
+  private refresh() {
+    this._refreshRequests++;
+  }
+
+  get walletManager() {
+    // trigger a refresh as we don't have access to the internal methods of the wallet manager.
+    this._refreshRequests;
```

Comment:
Ues mobx should notice a reference to the observable member here, if any observer-wrapped react component (may a provider) references `walletManager` during render.

Reply:
> Ues mobx should notice a reference to the observable member here, if any observer-wrapped react component (may a provider) references `walletManager` during render.
> 
> Mobx runs a function that tracks observables being referenced to figure out what observables need subscribing:
 https://github.dev/mobxjs/mobx/blob/faf075cc70a16a86443b24c8dd120516bac016ed/packages/mobx/src/core/derivation.ts#L167